fedimint-api/src/tiered_multi.rs

```diff
@@ -78,35 +86,49 @@ impl<T> TieredMulti<T> {
         tier_eq && per_tier_eq
     }
 
-    pub fn iter_tiers(&self) -> impl Iterator<Item = (&Amount, &Vec<T>)> {
+    /// Returns an borrowing iterator
+    pub fn iter(&self) -> impl Iterator<Item = (&Amount, &Vec<T>)> {
         self.0.iter()
     }
 
-    // Note: order of the elements is important here: from lowest tiers to highest, then in order of elements in the Vec
+    /// Returns an iterator over every `(Amount, &T)`
+    ///
+    /// Note: The order of the elements is important:
+    /// from the lowest tier to the highest, then in order of elements in the Vec
     pub fn iter_items(&self) -> impl Iterator<Item = (Amount, &T)> + DoubleEndedIterator {
+        // Note: If you change the method implementation, make sure that the returned order of the
+        // elements stays consistent.
         self.0
             .iter()
             .flat_map(|(amt, coins)| coins.iter().map(move |c| (*amt, c)))
     }
-    // Note: order of the elements is important here: from lowest tiers to highest, then in order of elements in the Vec
+
+    /// Returns an consuming iterator over every `(Amount, T)`
+    ///
+    /// Note: The order of the elements is important:
+    /// from the lowest tier to the highest, then in order of elements in the Vec
     pub fn into_iter_items(self) -> impl Iterator<Item = (Amount, T)> + DoubleEndedIterator {
+        // Note: If you change the method implementation, make sure that the returned order of the
+        // elements stays consistent.
         self.0
             .into_iter()
             .flat_map(|(amt, coins)| coins.into_iter().map(move |c| (amt, c)))
     }
 
-    /// Returns the max number of notes for any given denomination tier
-    pub fn max_tier_len(&self) -> usize {
+    /// Returns the maximum length of all tier vectors
+    pub fn max_value_len(&self) -> usize {
         self.0.values().map(|notes| notes.len()).max().unwrap_or(0)
     }
 
-    pub fn check_tiers<K>(&self, keys: &Tiered<K>) -> Result<(), InvalidAmountTierError> {
+    /// Verifies that all keys in `self` are present in the keys of the given parameter `Tiered`
+    pub fn tiers_exist_in<K>(&self, keys: &Tiered<K>) -> Result<(), InvalidAmountTierError> {
```

Comment:
Similar here.  `all_tier_exist_in` maybe?

Reply:
mh, I think the `all_tier` in `all_tier_exist_in` is represented with `tiers` in `tiers_exist_in`.  Or `all_tiers_exist_in` (note the `s`)

Reply:
Specifically it's a `any` vs `all` clarification.

Reply:
let's go for `all_tiers_exist_in` then